File: src/main/kotlin/dev/jefy/connectpro/management/appliacaion/command/BadgeCommand.java
package dev.jefy.connectpro.management.appliacaion.command;

import org.jspecify.annotations.NullMarked;

import dev.jefy.connectpro.management.appliacaion.dtos.BadgeRequest;
import dev.jefy.connectpro.management.domain.vo.BadgeId;

/**
 * @author Jôph Yamba
 */
@NullMarked
public interface BadgeCommand {
    BadgeId create(BadgeRequest request);
    BadgeId update(BadgeId badgeId, BadgeRequest request);
    void delete(BadgeId badgeId);
}

File: src/main/kotlin/dev/jefy/connectpro/management/appliacaion/command/BadgeCommand.kt
package dev.jefy.connectpro.management.appliacaion.command

import dev.jefy.connectpro.management.appliacaion.dtos.BadgeRequest
import dev.jefy.connectpro.management.domain.vo.BadgeId
import org.jspecify.annotations.NullMarked

/**
 * @author Jôph Yamba
 */
@NullMarked
interface BadgeCommand {
    fun create(request: BadgeRequest): BadgeId
    fun update(badgeId: BadgeId, request: BadgeRequest): BadgeId
    fun delete(badgeId: BadgeId)
}
